### ISSUE TITLE
aws_launch_configuration add support for Instance Metadata Options

### ISSUE THREAD

**Issue by jaysiyani** [hashicorp/terraform-provider-aws#14621]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently the aws_launch_configuration resource does not support Instance Metadata Options. These options can be required when using the Instance Metadata Service Version 2, which new versions of the AWS SDK now use:

Following can be done by aws-cli 

```
aws autoscaling create-launch-configuration \
  --launch-configuration-name my-lc-with-imdsv2 \
  --image-id ami-01e24be29428c15b2 \
  --instance-type t2.micro \
      ...    
  --metadata-options "HttpEndpoint=enabled,HttpTokens=required" 
```

### New or Affected Resource(s)

* aws_launch_configuration

### Potential Terraform Configuration

```hcl
resource "aws_launch_configuration" "foo" {
  name = "foo"
  metadata_options {
    http_endpoint               = "enabled"
    http_put_response_hop_limit = 2
    http_tokens                 = "required"
  }
}
```

### References

* https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-config.html
* https://github.com/aws/aws-sdk-go/blob/3386d6000934896486e4686f32d55a97abdea1a0/service/autoscaling/api.go#L7047

**Comment by jaysiyani** [comment]
(WIP) https://github.com/terraform-providers/terraform-provider-aws/pull/14637

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.22.0 of the Terraform AWS Provider, later this week. Thank you to @jaysiyani for the implementation. 👍
